### ISSUE TITLE
UX: Provide a list of experimental features in the Settings

### ISSUE THREAD

**Issue by slemeur** [containers/podman-desktop#10223]
### UX Description

In order to simplify the discoverability and management of the experimental features, we would like to introduce a list in the settings.

That list should be at the primary navigation level in the settings.
For the end-user the list must allow the following:
- Understand what the experimental feature enable and the use case it unlocks
- See the state of the experimental feature, if it's enable or disable
- Have an action enabling to share feedback

### Request type

Application mockups/designs

### Primary Contact

@slemeur

**Comment by slemeur** [comment]
Some mockups have already been provided by @MariaLeonova.

![Image](https://github.com/user-attachments/assets/928adefe-a3f8-4193-a2cf-2b164a0ddcdf)

We need to further refine the mockup.

**Navigation:**
- The list of experimental features should be at the top level otherwise people are going to miss them. That could probably go under settings

**List of experimental features:**
I think the option 1 is probably the best one, but would like to further iterate on it. 
- the "Share feedback" is too easy to miss
- the icon for "thumbs up" or "thumbs down" are not very clear.

Should sharing a feedback be a 2 steps flow? 
- First the user share a "lazy feedback"
![Image](https://github.com/user-attachments/assets/acc2fffe-9101-40cf-ae83-0dca04f07844)
- Second we bring a "share your thoughts and join the discussion" popup, where the can tell a bit more.

**Comment by MariaLeonova** [comment]
>The list of experimental features should be at the top level otherwise people are going to miss them. That could probably go under settings

Certainly can be! Should it come last on the list? The Setting items don't seem to be arranged in alphabetical order.

>Should sharing a feedback be a 2 steps flow?

My thoughts were: they click on the smiley / thumb up (down) /  share feedback link and the pop up comes up. In my experience, once I've selected some feedback, I get annoyed buy the suggestion to add more comments. It would be more honest to bring the whole pop up right away.

>the "Share feedback" is too easy to miss

I was following some of the existing patterns, such as: 
<img src="https://github.com/user-attachments/assets/45b49eb9-94d4-4f70-b07a-6ab5d7c7b7b4" width="500">
  
<img src="https://github.com/user-attachments/assets/7573fad0-1f29-4c5b-ad6a-3df3b4d7ae44" width="500">
    
<img src="https://github.com/user-attachments/assets/e218ff47-feec-462c-a917-83f6de986e10" width="500">

If those are not visible enough, the usual ways to **make UI elements stand out** are: 
- color
- size
- shape
- position

We could work with that.


All that said, here's the latest mock, taking your comments into consideration:


![Image](https://github.com/user-attachments/assets/45512c4e-1649-408e-b3ca-6f3d568efca8)

**Comment by slemeur** [comment]
> Should it come last on the list? The Setting items don't seem to be arranged in alphabetical order.

Yes, I think that would work. As you did in the mockup. Though I'm not sure we are technically able to add the "experiment" icon. 

I'm still a bit hesitant about the way we are enticing the user to share a feedback here. What could we do to make it more engaging and rewarding? Should we display the number of "+1" or "-1"? If this is linked to a github discussion, maybe we want to show the number of comments? 

Finally the title of the action, I think it should be "Send feedback". Could it be a real button? not a link?

**Comment by MariaLeonova** [comment]
I wonder if it's technically possible to pul the number of upvotes from the discussion page and show them here? 

To make it more rewarding there probably needs to be some mechanism of rewards :) Otherwise I would suggest to try and see if the mechanism works? 

For instance, are you happy with the current way to Give feedback in Podman Desktop? Or is there something that needs to be changed?

Maybe there's some way to acknowledge their feedback un the Discussions and thank the users; potentially highlight their insights in the newsletter / product updates?

**Comment by MariaLeonova** [comment]
I added a full button in Option 4, it does take quite a bit of space: 

![Image](https://github.com/user-attachments/assets/4bbbbd5c-4e90-4ceb-9e32-35bea568e3f6)

**Comment by MariaLeonova** [comment]
If we think about user behaviour, the list in Settings might not be the primary space to ask for feedback on experimental features. 
A better option would be to ask for it in context, which would be handled in https://github.com/podman-desktop/podman-desktop/issues/10228 and https://github.com/podman-desktop/podman-desktop/issues/10227.

By in context I mean a potential pop-up / email reminder / notification after they've been using it for some time.

@slemeur would all of these be implemented at once or gradually? I.e. would step 1 be adding the feature list to Settings, step 2 setting up feedback mechanism, step 3 creating GitHub discussions, step 4 setting up feedback reminder?

**Comment by MariaLeonova** [comment]
I am adding 2 more cards option: 
![Image](https://github.com/user-attachments/assets/d37ef8d8-3985-49cf-b52b-ab520579adff)

**Comment by MariaLeonova** [comment]
For discussions, should there be a new section on the [Disussions](https://github.com/podman-desktop/podman-desktop/discussions/new/choose) or could it go under **General** or **Q&A**? 

@benoitf

**Comment by benoitf** [comment]
we can create a dedicated section yes.

Also should it be prefixed by the version of Podman Desktop

like if there "Task Manager" experimental (it could be the one done in the first attempt) and now the second one and maybe there will be a third one, etc

How do we track ?

it's a single section and then we discuss all enhancements from there (about the task manager) or it's related to an iterator

**Comment by MariaLeonova** [comment]
@benoitf my first instinct is to track all iterations of a particular feature in one discussion thread, so if someone is looking for a specific comment or functionality, they don't have to look through a number of entries, and all info is in one place. I suppose not all the users will be familiar with versions, too. This is from **the user** POV. 

On the other hand, if it's easier for **the team** to track iterations separately, then the discussion can happen in a separate thread for a particular version. If we suppose, the feedback feature is mainly for the team to help in development, then we should consider what's best for that. 

Here's an example from GitHub Public Preview for issues feedback: https://github.com/orgs/community/discussions/139933 

@slemeur @Firewall what are your thoughts?

**Comment by Firewall** [comment]
> my first instinct is to track all iterations of a particular feature in one discussion thread, so if someone is looking for a specific comment or functionality, they don't have to look through a number of entries, and all info is in one place. I suppose not all the users will be familiar with versions, too. This is from the user POV.

+1 totally agree with that. 

> On the other hand, if it's easier for the team to track iterations separately, then the discussion can happen in a separate thread for a particular version. If we suppose, the feedback feature is mainly for the team to help in development, then we should consider what's best for that. 
Here's an example from GitHub Public Preview for issues feedback: https://github.com/orgs/community/discussions/139933

I think our goal is getting user feedback, we should prioritize making it as easy as possible for users giving us that feedback. So I would go with whatever is easier for the users. So that means I my vote goes to `1 feature, 1 thread, all iterations`

I really like how GitHub handles. I personally gave feedback that way

**Comment by MariaLeonova** [comment]
So looking at Github Discussions API's I think we have 2 options:

    
**1. Do our own feedback mechanism and link to GitHub Discussions from the UI.**
  
Given that GitHub discussions use thumbs up / thumbs down for +1 / -1, it would probably make sense to have that in the UI.
![Image](https://github.com/user-attachments/assets/2e78ef94-92bd-4d71-a142-1be55fc6ff2e)

Once a user clicks either one (thumbs up/down/send feedback), they will see a popup with an ability to provide more info and go to GitHub discussions to leave a comment. This pattern will directly mimic the existing feedback mechanism in Podman desktop. 

![Image](https://github.com/user-attachments/assets/fac74717-6b3f-4e46-86be-2288b242b478)



**2. Connect to GH discussions directly and pull the upvotes, number of comments and reactions from there.** 

<img width="1527" alt="Image" src="https://github.com/user-attachments/assets/f7784ec2-3060-47d5-8f94-616e7b45e8ee" />

**Comment by axel7083** [comment]
> 2. Connect to GH discussions directly and pull the upvotes, number of comments and reactions from there.

I have some doubt with this option because the user cannot interact with the upvote from podman desktop, it would have to open the discussion in the browser.

**Comment by MariaLeonova** [comment]
@slemeur @Firewall as we discussed, here's a proposal on how we could show number of comments and upvotes + invite users to join GitHub discussion: 
<img width="1117" alt="Image" src="https://github.com/user-attachments/assets/50c0282b-7343-4b22-a3ad-04c1e2327209" />

I am suspecting that people might get a bit frustrated, if they can't click on upvotes or comments, so if those could be linked to discussions - at least go there without any action even; that would be beneficial. So in the top card (docker compatibility) I left those as links. 

If there's no visual indication of an external link, a popup should come up as it does elsewhere in Podman Desktop; warning about leaving the app.

**Comment by axel7083** [comment]
Hi @MariaLeonova nice!

I like the second one `Join discussion on GitHub [icon-external]` (I like the external icon, as it visually show that this is external)

I like the third one for the arrow without border, as it feel less clickable: the first one (purple) feel clickable, where it is not (?) as this is readonly. 

I also wonder, if we should not just have the comment count ? I was wondering cc @slemeur don't we want to keep the 👍 👎 that will use our telemtry?

**Comment by MariaLeonova** [comment]
@slemeur please take a look at the latest proposal based on our December discussion  - I would love to know your thoughts!

**Comment by slemeur** [comment]
> I also wonder, if we should not just have the comment count ?

I think it's nice to show the two, "comment count" and "upvote"

I tend to agree with @axel7083:
- the action to engage the user to join the discussion should be clear and the icon is better conveying that.
- the way to display comment count and upvote is looking better without the border and without the primary color. (so option 3 works better)

A couple of other feedback:
- we should think about the tooltips behind the "comment count" and "upvote" 
- The label "Join discussion on GitHub" feels not very clear to convey the message that user can share their feedback. What about "Share your thoughts and join the discussion on GitHub"? or simply "Share your feedback" (still with the icon)?

**Comment by MariaLeonova** [comment]
![Image](https://github.com/user-attachments/assets/258dbd48-b50a-414c-b76f-d34553fd5f83)

**Comment by MariaLeonova** [comment]
Based on yesterday's UX call, adding a new suggestion:

![Image](https://github.com/user-attachments/assets/cfd0800d-ade9-449c-824e-cbab4c15935c)

1 - Grey label option
2 - Share feedback button (short)
3 - Purple label option
4 - Comments tooltip
5 - Upvotes tooltip

I also separated the feedback button and comments/upvotes into separate lines, in case we can't fetch the numbers; then they just wouldn't show. (or, alternatively, can say N/A).

Here's the same screen without the numbers:

![Image](https://github.com/user-attachments/assets/c68807f9-662d-4021-9d5c-57e993d35bd9)

**Comment by deboer-tim** [comment]
Looking nicer! A couple comments & nits:
- I don't see a difference between them... wrong second image?
- The label is nice, and makes them easy to identify. I prefer the purple one.
- The title line in existing preferences is small though, putting the label at the top would move the descriptions down and might add some unexpected padding.
- IMHO the comment/vote line should be to the right of the feedback link. Why? Think of a slow connection where it takes a second or two to get a response from GitHub, or there's a momentary delay. Each time one loads, all of the preferences below it would shift. It's better if the layout is always fixed, and data either appears or doesn't.

**Comment by slemeur** [comment]
Thanks @MariaLeonova ! 

When I see the screen, I'm thinking about what gets my attention at first. In the current mockup, the "experimental" labels are definitely getting my attention, but I don't think that's where we want to draw the attention in a first place.

I'm thinking about what we want the user to do.
1- Enabling / Disabling the exp feature. Here I think it's fairly clear and identifiable on the screen.
2- Engaging on sharing feedback. And on that point, I thing the hierarchy of information is incorrect. The "Share Feedback" as a link is too subttle and very easy to be missed. And now, with the experimental label, I think it's confusing even more the uer to understand of what the primary actions are.

So in general:
- For the label, the option 1, in gray is looking better, because that's going to be a repetitive label, not giving an information.
- Tooltips 4 and 5 are nice.


Can we look for an alternate button style for the "Share feedback" button?

**Comment by axel7083** [comment]
The experimental settings appear in two places, the `Preferences` and in the `Experimental page`.

During the UX call I mention that we should have a way to differentiate the experimental option in the preference from the others, so the label may be good just for the preferences page, and not needed for the experimental page, as it would be repetitive ?

**Comment by MariaLeonova** [comment]
> The experimental settings appear in two places, the `Preferences` and in the `Experimental page`.
> 
> During the UX call I mention that we should have a way to differentiate the experimental option in the preference from the others, so the label may be good just for the preferences page, and not needed for the experimental page, as it would be repetitive ?

Good point - I wonder if it's worth keeping it there for consistency?

**Comment by MariaLeonova** [comment]
Thank you, all, for feedback - I put two more options side by side. In (1) the button is primary + icon, so it REALLY stands out, might be even a little too much?

In (2) the Experimental label is underneath the Title and the button is not as prominent. 

In both I shortened the button text to just say **Share feedback**. In case we'd want to elaborate on that, there can be a tooltip to specify, where the will be able to do that. 

<img width="1599" alt="Image" src="https://github.com/user-attachments/assets/e9573802-fd5b-4809-b6ce-f8b6cd1a2736" />

**Comment by slemeur** [comment]
I agree that the button in the option 1 is standing out quite a bit, but I think that is clear to the end-user and it clearly show the action the user can do.

**Comment by slemeur** [comment]
Closing as solved! 
Good job @MariaLeonova !
